shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm

```diff
@@ -32,8 +32,12 @@ - (instancetype)init {
         displayLinkWithTarget:self
                      selector:@selector(onDisplayLink:)] retain];
     _displayLink.paused = YES;
-    [_displayLink addToRunLoop:[NSRunLoop currentRunLoop]
-                       forMode:NSRunLoopCommonModes];
+
+    blink::Threads::UI()->PostTask([client = [self retain]]() {
+      [client->_displayLink addToRunLoop:[NSRunLoop currentRunLoop]
+                                 forMode:NSRunLoopCommonModes];
```

Comment:
Though I am sure I don't need an implicit ARP activation here because I am managing my own ref counts on the client, I can also be safe and add an explicit activation to guard us against autoreleased objects within `addToRunLoop:forMode:`. If that makes you more comfortable.